File: C_Programming/Practise_Programs/PP_167.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_167
//
//  Description       : This program toggles the 4th bit of a given
//                      integer number using the bitwise XOR operator.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 20/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UINT;

//////////////////////////////////////////////////////////////////
//
//  Function Name     : main()
//
//  Description       : It accepts an integer number from the user
//                      and toggles its 4th bit using the bitwise
//                      XOR operator with an appropriate mask.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 20/08/2026
//
//////////////////////////////////////////////////////////////////

int main()
{
    UINT iNo = 0;
    UINT iMask = 0;

    printf("Enter Number : \n");
    scanf("%u", &iNo);

    iMask = 0x00000008;

    iNo = iNo ^ iMask;
    
    printf("Updated number : %u\n", iNo);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      10
//
//  Sample Output :
//
//      Enter Number :
//      Updated number : 2
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_167.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_167
//
//  Description       : This program accepts the number of rows
//                      and columns from the user and displays
//                      a diagonal star pattern using nested
//                      for loops.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 04/09/2026
//
//  Time Complexity   : O(n * m)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class Pattern
{
    public static void Display(int iRow, int iCol)
    {
        int i, j = 0;

        for(i = iRow; i >= 1; i--)
        {
            for(j = 1; j <= iCol; j++)
            {
                if(i == j)
                {
                    System.out.print("* \t");
                }
                else
                {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }

        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                if(i == j)
                {
                    System.out.print("* \t");
                }
                else
                {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}

class PP_167
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1, iValue2 = 0;

        System.out.println("Enter the number of Rows : ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter the numbe of Columns : ");
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.Display(iValue1, iValue2);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the number of Rows : 5
//      Enter the number of Columns : 5
//
//  Sample Output :
//
//      *    
//       *   
//        *  
//         * 
//          *
//          *
//         *  
//        *   
//       *    
//      *     
//
//////////////////////////////////////////////////////////////////